File: project-task-management-ui/src/app/core/models/project.ts
import { Task } from "./task";

export enum ProjectStatus {
  ACTIVE = 'Active',
  ON_HOLD = 'OnHold',
  COMPLETED = 'Completed'
}

export interface Project {
  id: number;
  tilte: string;
  description?: string;
  startDate: string;
  endDate: string;
  status: string;
  priority: string;
  projectManagerName?: string;
  progress: number;
  members?: string[];
  createdAt: string;
  tasksCount: number;
  tasks?: Task[];
}

File: ProjectTaskManagement.API/ProjectTaskManagement.Data/Entities/Project.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace ProjectTaskManagement.Data.Entities
{
    public class Project
    {
        public int Id { get; set; }

        public string Title { get; set; } = null!;
        public string? Description { get; set; }

        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }

        public string Status { get; set; } = "NotStarted";
        public string Priority { get; set; } = "MEDIUM";
        public string? ProjectManagerName { get; set; }
        public string? Members { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        // العلاقة
        public List<ProjectTask> Tasks { get; set; } = new();


    }
}
